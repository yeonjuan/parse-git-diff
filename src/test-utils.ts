import Context from './context';

type Validator<Args extends any[] = any[]> = (...args: Args) => boolean;

function createBooleanTester<V extends Validator>(
  validator: V,
  result: boolean
) {
  return function validTester(cases: Parameters<V>[]): void {
    cases.forEach((c) => {
      expect(validator(...c)).toBe(result);
    });
  };
}

export function createValidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, true);
}

export function createInvalidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, false);
}

export function createContext(diff: string, initial: number = 1) {
  const context = new Context(diff);
  for (let i = 0; i < initial - 1; i++) {
    context.nextLine();
  }
  return context;
}
