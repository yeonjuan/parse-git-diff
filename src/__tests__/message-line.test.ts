import { getFixture } from './test-utils';
import parseGitDiff from '../parse-git-diff';

describe('message-line', () => {
  const fixture = getFixture('message-line');

  it('parse `message-line`', () => {
    expect(parseGitDiff(fixture)).toMatchSnapshot();
  });
});
