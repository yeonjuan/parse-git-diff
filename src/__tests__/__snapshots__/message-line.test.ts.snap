// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message-line parse \`message-line\` 1`] = `
{
  "files": [
    {
      "chunks": [
        {
          "changes": [
            {
              "content": "newfile",
              "lineAfter": 1,
              "lineBefore": 1,
              "type": "UnchangedLine",
            },
            {
              "content": "newline",
              "lineBefore": 2,
              "type": "DeletedLine",
            },
            {
              "content": "newline",
              "lineAfter": 2,
              "type": "AddedLine",
            },
            {
              "content": "No newline at end of file",
              "type": "MessageLine",
            },
          ],
          "fromFileRange": {
            "lines": 2,
            "start": 1,
          },
          "toFileRange": {
            "lines": 1,
            "start": 1,
          },
          "type": "Chunk",
        },
      ],
      "path": "rename.md",
      "type": "ChangedFile",
    },
    {
      "chunks": [
        {
          "changes": [
            {
              "content": "newfile2",
              "lineAfter": 1,
              "lineBefore": 1,
              "type": "UnchangedLine",
            },
            {
              "content": "newline2",
              "lineBefore": 2,
              "type": "DeletedLine",
            },
            {
              "content": "newline2",
              "lineAfter": 2,
              "type": "AddedLine",
            },
            {
              "content": "No newline at end of file",
              "type": "MessageLine",
            },
          ],
          "fromFileRange": {
            "lines": 2,
            "start": 1,
          },
          "toFileRange": {
            "lines": 1,
            "start": 1,
          },
          "type": "Chunk",
        },
      ],
      "path": "rename2.md",
      "type": "ChangedFile",
    },
  ],
  "type": "GitDiff",
}
`;
